File: impl/app/android/app/src/main/java/com/ak/contexta/ui/components/ArticleCard.kt
package com.ak.contexta.ui.components

import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.text.style.TextOverflow
import androidx.compose.ui.unit.dp
import com.ak.contexta.ui.theme.Muted
import com.ak.contexta.ui.theme.MutedSoft

data class ArticleCardData(
    val id: Long,
    val title: String?,
    val description: String,
    val difficultyLabel: String,
    val categoryLabel: String,
    val isReadCompleted: Boolean = false
)

@Composable
fun ArticleCard(
    article: ArticleCardData,
    onClick: () -> Unit,
    modifier: Modifier = Modifier
) {
    AppCard(
        onClick = onClick,
        modifier = modifier.fillMaxWidth()
    ) {
        Column(modifier = Modifier.fillMaxWidth()) {
            Text(
                text = article.title ?: article.description.take(40),
                style = MaterialTheme.typography.headlineMedium,
                maxLines = 1,
                overflow = TextOverflow.Ellipsis
            )
            Spacer(modifier = Modifier.height(4.dp))
            Text(
                text = article.description,
                style = MaterialTheme.typography.bodyMedium,
                color = Muted,
                maxLines = 2,
                overflow = TextOverflow.Ellipsis
            )
            Spacer(modifier = Modifier.height(8.dp))
            Row(
                horizontalArrangement = Arrangement.spacedBy(8.dp),
                verticalAlignment = Alignment.CenterVertically
            ) {
                AppBadge(
                    text = article.difficultyLabel,
                    variant = when (article.difficultyLabel) {
                        "CET4" -> AppBadgeVariant.Coral
                        "CET6" -> AppBadgeVariant.Green
                        else -> AppBadgeVariant.Default
                    }
                )
                Text(
                    text = article.categoryLabel,
                    style = MaterialTheme.typography.bodySmall,
                    color = Muted
                )
                if (article.isReadCompleted) {
                    Spacer(modifier = Modifier.weight(1f))
                    Text(
                        text = "✓ 已读",
                        style = MaterialTheme.typography.labelSmall,
                        color = MutedSoft
                    )
                }
            }
        }
    }
}

File: impl/app/flutter/lib/core/components/article_card.dart
import 'package:flutter/material.dart';

import '../theme/app_colors.dart';
import '../theme/app_dimens.dart';
import '../theme/app_type.dart';
import 'app_badge.dart';
import 'app_card.dart';

class ArticleCardData {
  const ArticleCardData({
    required this.id,
    required this.title,
    required this.description,
    required this.difficultyLabel,
    required this.categoryLabel,
    this.isReadCompleted = false,
  });

  final int id;
  final String? title;
  final String description;
  final String difficultyLabel;
  final String categoryLabel;
  final bool isReadCompleted;
}

/// 文章卡片（基于 AppCard）：serif 标题单行省略 + Muted 描述两行省略 +
/// 难度徽标（CET4 → Coral、CET6 → Green、其他 → Default）+ 分类 + 已读标记。
/// 对照 Kotlin ui/components/ArticleCard.kt。
class ArticleCard extends StatelessWidget {
  const ArticleCard({
    super.key,
    required this.article,
    required this.onClick,
  });

  final ArticleCardData article;
  final VoidCallback onClick;

  @override
  Widget build(BuildContext context) {
    return AppCard(
      onClick: onClick,
      child: Column(
        crossAxisAlignment: CrossAxisAlignment.start,
        children: [
          Text(
            article.title ?? (article.description.length > 40
                ? article.description.substring(0, 40)
                : article.description),
            style: AppType.textTheme.headlineMedium,
            maxLines: 1,
            overflow: TextOverflow.ellipsis,
          ),
          const SizedBox(height: AppSpacing.xxs),
          Text(
            article.description,
            style: AppType.textTheme.bodyMedium?.copyWith(
              color: AppColors.muted,
            ),
            maxLines: 2,
            overflow: TextOverflow.ellipsis,
          ),
          const SizedBox(height: AppSpacing.xs),
          Row(
            children: [
              AppBadge(
                article.difficultyLabel,
                variant: switch (article.difficultyLabel) {
                  'CET4' => AppBadgeVariant.coral,
                  'CET6' => AppBadgeVariant.green,
                  _ => AppBadgeVariant.default_,
                },
              ),
              const SizedBox(width: AppSpacing.xs),
              Text(
                article.categoryLabel,
                style: AppType.textTheme.bodySmall?.copyWith(
                  color: AppColors.muted,
                ),
              ),
              if (article.isReadCompleted) ...[
                const Spacer(),
                Text(
                  '✓ 已读',
                  style: AppType.textTheme.labelSmall?.copyWith(
                    color: AppColors.mutedSoft,
                  ),
                ),
              ],
            ],
          ),
        ],
      ),
    );
  }
}
